motoman_driver/src/joint_trajectory_streamer.cpp

```diff
@@ -401,6 +401,74 @@ void MotomanJointTrajectoryStreamer::streamingThread()
         }
       }
       break;
+
+    case TransferStates::POINT_STREAMING:
+      // if no points in queue, streaming complete, set to idle.
+      if (this->streaming_queue_.empty())
+      {
+        if (this->time_since_last < this->point_streaming_timeout)
+        {
+          time_since_last = ros::Time::now().toSec() - time_of_last;
+          ros::Duration(0.005).sleep();
+          //ROS_INFO("Time since last point: %f", time_since_last);
+          break;
+        }
+        else
+        {
+          ROS_INFO("Point streaming complete,  setting state to IDLE");
+          this->state_ = TransferStates::IDLE;
+          break;
+        }
+      }
+      // if not connected, reconnect.
+      if (!this->connection_->isConnected())
```

Comment:
Do you know why this reconnect logic is needed here?  Why doesn't the reconnect [here](https://github.com/ros-industrial/motoman/pull/88/files#diff-a99f5449c5369b017cba549fc88643c2R329) work?